feat: perps order creation

abretonc7s opened the pull request:
<!--
  Please submit this PR as a draft initially.
  Do not mark it as "Ready for review" until the template has been completely filled
  out, and PR status checks have passed at least once.
  -->

  ## **Description**

  This PR implements three key user stories for Perps trading functionality:

  1. **TAT-1022**: Users can now open market orders using their perp account balance
  2. **TAT-1023**: Take Profit/Stop Loss (TP/SL) functionality has been implemented to
  help users manage investment risk
  3. **TAT-1031**: Users can now open limit orders with proper price setting

  ## **Changelog**

  CHANGELOG entry: Added Perps market and limit order creation with Take Profit/Stop
  Loss functionality

  ## **Related issues**

  Fixes: TAT-1022, TAT-1023, TAT-1031

  ## **Manual testing steps**

  ### Market Orders (TAT-1022):
  1. Navigate to Perps trading view
  2. Ensure "Market" order type is selected
  3. Enter order amount using slider or direct input
  4. Verify order uses perp account balance
  5. Place market order successfully

  ### Take Profit/Stop Loss (TAT-1023):
  1. In order creation view, tap on "Take Profit" or "Stop Loss"
  2. Verify new dual input design (USD and percentage fields)
  3. Test that changing USD updates percentage and vice versa
  4. Verify percentage quick buttons (+5%, +10%, +15%, +20%) work correctly
  5. Set TP/SL values and confirm they're applied to the order

  ### Limit Orders (TAT-1031):
  1. Select "Limit" order type
  2. Verify limit price shows "Set price" placeholder (not current price)
  3. Tap to set limit price
  4. Test -1% and -2% buttons adjust the entered limit price
  5. Try to place order without limit price - should show validation error
  6. Set limit price and successfully place order

  ## **Screenshots/Recordings**

  ### **Before**
  - No perps order creation functionality
  - No TP/SL risk management
  - No limit order support

  ### **After**
  - Full market order creation with perp account balance
  - Comprehensive TP/SL interface with dual inputs
  - Limit order creation with proper UX and validation


LONG BTC:
https://github.com/user-attachments/assets/7b35b2e9-a405-44ef-bd7d-383589cd03a7

SHORT ETH:
![long_eth](https://github.com/user-attachments/assets/d0fd86e1-ea93-47e5-9157-9f832df9f3eb)


  ## **Pre-merge author checklist**

  - [x] I've followed [MetaMask Contributor 
  Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding 
  Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CO
  DING_GUIDELINES.md).
  - [x] I've completed the PR template to the best of my ability
  - [x] I've included tests if applicable
  - [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
  - [x] I've applied the right labels on the PR (see [labeling guidelines](https://githu
  b.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
  Not required for external contributors.

  ## **Pre-merge reviewer checklist**

  - [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code
  being changed).
  - [ ] I confirm that this PR addresses all acceptance criteria described in the ticket
   it closes and includes the necessary testing evidence such as recordings and or
  screenshots.

abretonc7s committed:
feat: basic deposit screens

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/1202-deposit

abretonc7s committed:
feat: working token selector

abretonc7s committed:
feat: ui cleaning

abretonc7s committed:
unit tests

abretonc7s committed:
unit tests

abretonc7s committed:
cleanup

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/1202-deposit

abretonc7s committed:
fix: jest snapshots

abretonc7s committed:
cleanup

abretonc7s committed:
formatting

abretonc7s committed:
i18n errors

abretonc7s committed:
feat: unit test coverage

abretonc7s committed:
Merge branch 'origin/main' into feat/1202-deposit

Resolved merge conflicts in:
- app/components/UI/Perps/routes/index.tsx
- app/components/UI/Perps/Views/PerpsView.tsx
- app/components/UI/Perps/services/HyperLiquidClientService.ts
- locales/languages/en.json

Combined deposit and positions functionality from both branches.

abretonc7s committed:
pr comments

abretonc7s committed:
pr comments

abretonc7s committed:
wip

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/1202-deposit

abretonc7s committed:
revert build changes

abretonc7s committed:
feat: working quotation

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/1202-deposit

abretonc7s committed:
fix: unit tests

abretonc7s committed:
fix: missing translations

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/1202-deposit

abretonc7s committed:
pr comments

abretonc7s committed:
pr comments

abretonc7s committed:
pr comments

abretonc7s committed:
fpr,attomg

abretonc7s committed:
wip

abretonc7s committed:
coverage and slippage modal

gambinish committed:
chore: Format

gambinish committed:
fix: Unit test PerpsLoader

gambinish committed:
fix: Centralize testIds

gambinish committed:
fix: PerpsDepositProcessingView.test.tsx

gambinish committed:
fix: PerpsDepositSuccessView.test.tsx PerpsQuoteDetailsCard/PerpsQuoteDetailsCard.test.tsx

gambinish committed:
fix: PerpsPayWithRow.test.tsx

gambinish committed:
chore: Lint:tsc

gambinish committed:
Merge branch 'main' into feat/1202-deposit

gambinish committed:
chore: Update Engine unit test

gambinish committed:
fix: Lint

gambinish committed:
chore: Update unit test

gambinish committed:
chore: Update snapshot

gambinish committed:
chore: Unit tests PerpsTokenSelector.test.tsx

gambinish committed:
chore: Bump unit coverage on new PerpsController methods

gambinish committed:
Merge branch 'main' into feat/1202-deposit

gambinish committed:
chore: Bump test coverage

gambinish committed:
Merge branch 'feat/1202-deposit' of github.com:MetaMask/metamask-mobile into feat/1202-deposit

gambinish committed:
fix: Lint tsc

gambinish committed:
chore: Bump unit test coverage

gambinish committed:
Merge branch 'main' into feat/1202-deposit

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into feat/1202-deposit

abretonc7s committed:
fix: linting

abretonc7s committed:
wip

abretonc7s committed:
merge main into branch

abretonc7s committed:
feat: working order

abretonc7s committed:
linting and unit test

abretonc7s committed:
price limit modal

abretonc7s committed:
live mid prices

abretonc7s committed:
styling

abretonc7s committed:
feat: auto peprs reconnect

abretonc7s committed:
fix: network icon on testnet deposit

abretonc7s committed:
missing i18n

abretonc7s committed:
cleanup

abretonc7s committed:
feat: restore perps manager

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
tpsl

abretonc7s committed:
price limit ux

abretonc7s committed:
revert: remove accidental build.sh changes from branch

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
formatting

abretonc7s committed:
unit tests

abretonc7s committed:
fix: tests setup

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

gambinish commented on app/components/UI/Perps/Views/PerpsDepositAmountView/PerpsDepositAmountView.tsx:
```diff
@@ -243,7 +239,7 @@ const PerpsDepositAmountView: React.FC<PerpsDepositAmountViewProps> = () => {
     }
 
-    return '-';
+    return '0x0';
```
This looks like a magic string. Can we define this in a variable?

gambinish commented on app/components/UI/Perps/Views/PerpsOrderView/PerpsOrderView.test.tsx:
```diff
@@ -0,0 +1,342 @@
+    const { TouchableOpacity, Text } = jest.requireActual('react-native');
+    return (
+      <TouchableOpacity
+        testID="perps-token-selector"
+        onPress={() => onSelectToken({ symbol: 'ETH', address: '0x123' })}
+      >
+        <Text>ETH</Text>
+      </TouchableOpacity>
+    );
+  },
+}));
+
+jest.mock('../../components/PerpsAmountDisplay', () => 'PerpsAmountDisplay');
+jest.mock('../../components/PerpsSlider', () => ({
+  __esModule: true,
+  default: ({
+    value,
+  }: {
+    value: number;
+    onValueChange: (v: number) => void;
+  }) => {
+    const { View } = jest.requireActual('react-native');
+    return <View testID="perps-slider" data-value={value} />;
+  },
+}));
+
+// Mock utility functions
+jest.mock('../../utils/tokenIconUtils', () => ({
+  enhanceTokenWithIcon: jest.fn((token) => token),
+}));
+
+jest.mock('../../utils/formatUtils', () => ({
+  formatPrice: jest.fn((price) => `$${price}`),
+}));
+
+// Mock network utils
+jest.mock('../../../../../util/networks', () => ({
+  getDefaultNetworkByChainId: jest.fn(() => ({ name: 'Arbitrum' })),
+  getNetworkImageSource: jest.fn(() => ({ uri: 'network-icon' })),
+}));
```
Similarly, I think these mocks might be overly aggressive. I don't think we need to mock the components that this View consumes, because we want to be testing the components themselves.

I also don't think the utils need to be mocked to get these tests passing since these are local utilities. For  external libraries like react navigation, mocking makes a lot more sense IMO.

I can help taking a look at this if you want.

gambinish commented on app/components/UI/Perps/Views/PerpsOrderView/PerpsOrderView.tsx:
```diff
@@ -0,0 +1,1037 @@
+  const { openTooltipModal } = useTooltipModal();
+
+  // Selectors
+  const tokenList = useSelector(selectTokenList);
```
I'm leaving a note here to double check that this token list is the correct one to be using. Do we need to consider multichain tokens like SOL?

abretonc7s replied:
En fait le paywith risque de sauter completement... pour v1

abretonc7s replied:
Actually it seems for V1 the PayWith may be fully removed which would simplify..

gambinish commented on app/components/UI/Perps/Views/PerpsOrderView/PerpsOrderView.tsx:
```diff
@@ -0,0 +1,1037 @@
+        });
+
+        navigation.navigate(Routes.PERPS.ORDER_SUCCESS, {
+          orderId: result.orderId || 'unknown',
```
Is there a case where an order doesn't have an orderId? Might make sense to handle the `orderId === 'unknown'` case separately

gambinish commented on app/components/UI/Perps/Views/PerpsView.tsx:
```diff
@@ -76,6 +139,9 @@ const PerpsView: React.FC<PerpsViewProps> = () => {
   const [result, setResult] = useState<string>('');
 
+  // Popular trading pairs
+  const POPULAR_ASSETS = ['BTC', 'ETH', 'SOL', 'ARB'];
```
Can we put this in our HyperliquidConfig?

gambinish commented on app/components/UI/Perps/components/PerpsLeverageBottomSheet/PerpsLeverageBottomSheet.styles.ts:
```diff
@@ -0,0 +1,148 @@
+      paddingTop: 16,
+      paddingBottom: 24,
+    },
+    leverageText: {
+      fontSize: 48,
+      fontWeight: '600',
+      lineHeight: 56,
+    },
+    leverageTextLow: {
+      color: colors.text.default,
+    },
+    leverageTextMedium: {
+      color: colors.warning.default,
+    },
+    leverageTextHigh: {
+      color: colors.error.default,
+    },
```
I think we need to be careful with styling text directly in stylesheets. My understanding is that we should rely on the `TextVariant` property of the `Text` component so that the design system team's updates will automatically propagate to whatever we make.

I think that the style sheet here should mostly be related to alignment/padding/margin

gambinish commented on app/components/UI/Perps/components/PerpsLimitPriceBottomSheet/PerpsLimitPriceBottomSheet.tsx:
```diff
@@ -0,0 +1,286 @@
+  }));
+
+  // Update price snapshot only when bottom sheet opens
+  useEffect(() => {
+    if (isVisible) {
+      const current = currentPriceData?.price
+        ? parseFloat(currentPriceData.price)
+        : passedCurrentPrice;
+      const mark = currentPriceData?.markPrice
+        ? parseFloat(currentPriceData.markPrice)
+        : current;
+      const bid = currentPriceData?.bestBid
+        ? parseFloat(currentPriceData.bestBid)
+        : current
+        ? current * DEFAULT_BID_MULTIPLIER
+        : undefined;
+      const ask = currentPriceData?.bestAsk
+        ? parseFloat(currentPriceData.bestAsk)
+        : current
+        ? current * DEFAULT_ASK_MULTIPLIER
+        : undefined;
+
+      setPriceSnapshot({
+        currentPrice: current,
+        markPrice: mark,
+        bestBid: bid,
+        bestAsk: ask,
+      });
+    }
+  }, [isVisible]); // eslint-disable-line react-hooks/exhaustive-deps
```
Also, don't we also want to recompute the price snapshot if `currentPriceData` changes? We may want more dependencies in this `useEffect`

abretonc7s replied:
We need to have a snapshot of the price at a given point, I initially added the live prices in the component but it was confusing as the prices were always changing. Eventually I kept it as is, we can discuss and adapt as we try

gambinish commented on app/components/UI/Perps/components/PerpsOrderHeader/PerpsOrderHeader.tsx:
```diff
@@ -0,0 +1,146 @@
+  TextColor,
+} from '../../../../../component-library/components/Texts/Text';
+import TokenIcon from '../../../Swaps/components/TokenIcon';
+// import Routes from '../../../../../constants/navigation/Routes'; // No longer needed
+import { formatPrice, formatPercentage } from '../../utils/formatUtils';
+import { HYPERLIQUID_ASSET_ICONS_BASE_URL } from '../../constants/hyperLiquidConfig';
+// import type { PerpsNavigationParamList } from '../../controllers/types'; // No longer needed
```
Cleanup: Can remove these commented lines if they aren't being used.

gambinish commented on app/components/UI/Perps/components/PerpsOrderHeader/PerpsOrderHeader.tsx:
```diff
@@ -0,0 +1,146 @@
+            style={styles.tokenIcon}
+          />
+          <Text variant={TextVariant.HeadingMD} style={styles.headerTitle}>
+            {asset} {price > 0 ? formatPrice(price) : '$---'}
```
This seems like it's a bit brittle. Can we put this in a function to catch some edge cases?

gambinish commented on app/components/UI/Perps/components/PerpsSlider/PerpsSlider.tsx:
```diff
@@ -0,0 +1,258 @@
+      const percentage = (value - minimumValue) / (maximumValue - minimumValue);
+      const newPosition = percentage * widthRef.current;
+      translateX.value = withSpring(newPosition, {
+        damping: 15,
+        stiffness: 400,
```
Can probably put this into a config.

gambinish commented on app/components/UI/Perps/components/PerpsSlider/PerpsSlider.tsx:
```diff
@@ -0,0 +1,258 @@
+  // Convert position to value
+  const positionToValue = useCallback(
+    (position: number, width: number) => {
+      'worklet';
```
What is this? Is this required?

gambinish commented on app/components/UI/Perps/components/PerpsTPSLBottomSheet/PerpsTPSLBottomSheet.tsx:
```diff
@@ -0,0 +1,528 @@
+  };
+
+  // Quick percentage buttons
+  const takeProfitPercentages = [5, 10, 15, 20];
+  const stopLossPercentages = [5, 10, 15, 20];
```
Can probably put this into HyperliquidConfig

gambinish commented on app/components/UI/Perps/components/PerpsTPSLBottomSheet/PerpsTPSLBottomSheet.tsx:
```diff
@@ -0,0 +1,528 @@
+                style={styles.input}
+                value={takeProfitPercentage}
+                onChangeText={(text) => {
+                  // Allow only numbers and decimal point
+                  const sanitized = text.replace(/[^0-9.]/g, '');
+                  // Prevent multiple decimal points
+                  const parts = sanitized.split('.');
+                  if (parts.length > 2) return;
+                  setTakeProfitPercentage(sanitized);
+
+                  // Update price based on percentage
+                  if (sanitized && !isNaN(parseFloat(sanitized))) {
+                    const price = calculatePriceForPercentage(
+                      parseFloat(sanitized),
+                      true,
+                    );
+                    setTakeProfitPrice(price);
+                    setSelectedTpPercentage(parseFloat(sanitized));
+                  } else {
+                    setTakeProfitPrice('');
+                    setSelectedTpPercentage(null);
+                  }
+                }}
```
It looks like much of this logic is repeated from above. Can we put this into the same memoized callback and handle the state updates separately?

gambinish commented on app/components/UI/Perps/components/PerpsTPSLBottomSheet/PerpsTPSLBottomSheet.tsx:
```diff
@@ -0,0 +1,528 @@
+              <TextInput
+                style={styles.input}
+                value={stopLossPercentage}
+                onChangeText={(text) => {
+                  // Allow only numbers and decimal point
+                  const sanitized = text.replace(/[^0-9.]/g, '');
+                  // Prevent multiple decimal points
+                  const parts = sanitized.split('.');
+                  if (parts.length > 2) return;
+                  setStopLossPercentage(sanitized);
+
+                  // Update price based on percentage
+                  if (sanitized && !isNaN(parseFloat(sanitized))) {
+                    const price = calculatePriceForPercentage(
+                      parseFloat(sanitized),
+                      false,
+                    );
+                    setStopLossPrice(price);
+                    setSelectedSlPercentage(parseFloat(sanitized));
+                  } else {
+                    setStopLossPrice('');
+                    setSelectedSlPercentage(null);
+                  }
+                }}
```
Same thing here about memoizing a callback to be reused

gambinish commented on app/components/UI/Perps/components/PerpsTPSLBottomSheet/PerpsTPSLBottomSheet.tsx:
```diff
@@ -0,0 +1,528 @@
+                  }
+                }}
+              />
+              <Text variant={TextVariant.BodyMD} color={TextColor.Alternative}>
+                USD
+              </Text>
```
Note: It seems like we are hardcoding USD in several places in the Perps directory. We may want to consider having a separate cleanup PR that consolidates these all into a variable definition

gambinish commented on app/components/UI/Perps/components/PerpsTPSLBottomSheet/PerpsTPSLBottomSheet.tsx:
```diff
@@ -0,0 +1,528 @@
+              <TextInput
+                style={styles.input}
+                value={stopLossPrice}
+                onChangeText={(text) => {
+                  // Allow only numbers and decimal point
+                  const sanitized = text.replace(/[^0-9.]/g, '');
+                  // Prevent multiple decimal points
+                  const parts = sanitized.split('.');
+                  if (parts.length > 2) return;
+                  setStopLossPrice(sanitized);
+
+                  // Update percentage based on price
+                  if (sanitized) {
+                    const percentage = calculatePercentageForPrice(
+                      sanitized,
+                      false,
+                    );
+                    setStopLossPercentage(percentage);
+                  } else {
+                    setStopLossPercentage('');
+                  }
+                  setSelectedSlPercentage(null);
+                }}
+                placeholder="0"
+                placeholderTextColor={colors.text.muted}
+                keyboardType="numeric"
+                onFocus={() => setSlPriceInputFocused(true)}
+                onBlur={() => {
+                  setSlPriceInputFocused(false);
+                  // Format on blur if there's a value
+                  if (stopLossPrice && !isNaN(parseFloat(stopLossPrice))) {
+                    setStopLossPrice(formatPrice(stopLossPrice));
+                  }
+                }}
```
Same thing here about the memoized callback

gambinish commented on app/components/UI/Perps/hooks/usePerpsPaymentTokens.ts:
```diff
@@ -0,0 +1,156 @@
+  }, [networkConfigurations]);
+
+  // Get all tokens with balances across networks
+  const tokensWithBalance = useTokensWithBalance({ chainIds: allChainIds });
```
We want to be careful that this doesn't brick performance of this hook.

abretonc7s replied:
This will also be removed completely from yesterday's meeting with confirmation, it seems we will need different flow

gambinish commented on app/components/UI/Perps/services/PerpsConnectionManager.ts:
```diff
@@ -0,0 +1,123 @@
+  }
+
+  async disconnect(): Promise<void> {
+    this.connectionRefCount--;
+    DevLogger.log(
+      `PerpsConnectionManager: Disconnection requested (refCount: ${this.connectionRefCount})`,
+    );
+
+    // Only disconnect when all references are gone
+    if (this.connectionRefCount <= 0) {
+      this.connectionRefCount = 0; // Ensure it doesn't go negative
+
+      if (this.isConnected || this.isInitialized) {
+        try {
+          DevLogger.log(
+            'PerpsConnectionManager: Disconnecting (no more references)',
+          );
+          // Reset state before disconnecting to prevent race conditions
+          this.isConnected = false;
+          this.isInitialized = false;
+          this.isConnecting = false;
+
+          await Engine.context.PerpsController.disconnect();
+        } catch (error) {
+          DevLogger.log('PerpsConnectionManager: Disconnection error', error);
+        }
+      }
+    }
+  }
+
+  getConnectionState() {
+    return {
+      isConnected: this.isConnected,
+      isConnecting: this.isConnecting,
+      isInitialized: this.isInitialized,
+    };
+  }
+}
+
+export const PerpsConnectionManager = PerpsConnectionManagerClass.getInstance();
```
Can we add unit tests here?

abretonc7s commented on app/components/UI/Perps/Views/PerpsView.tsx:
```diff
@@ -222,9 +291,17 @@ const PerpsView: React.FC<PerpsViewProps> = () => {
+              </Text>
+              {parseFloat(cachedAccountState.totalBalance) === 0 && (
+                <Text variant={TextVariant.BodySM} color={TextColor.Warning}>
+                  No funds deposited. Use &apos;Deposit Funds&apos; to get
+                  started.
```
These were just added for quick testing, will be removed probably from your PR

abretonc7s committed:
pr comments

abretonc7s committed:
pr comments

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
test coverage

abretonc7s committed:
formatting

abretonc7s committed:
fix: unit tests

abretonc7s committed:
formatting

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup tests

abretonc7s committed:
formatting

abretonc7s committed:
coverage

abretonc7s committed:
coverage

abretonc7s committed:
coverage

abretonc7s committed:
pr comments

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
fix: remove accidentally committed e2e test file

Remove permission-system-update-permissions.spec.ts that was committed by mistake

abretonc7s committed:
Revert "fix: remove accidentally committed e2e test file"

This reverts commit b8e4658c6787a21c9cbf541633add60e70a2f003.

abretonc7s committed:
fix: revert unintended changes to e2e test file

Restore permission-system-update-permissions.spec.ts to its original state before formatting changes

abretonc7s committed:
fix: unit test

abretonc7s commented on app/components/UI/Perps/Views/PerpsOrderView/PerpsOrderView.tsx:
```diff
@@ -0,0 +1,1037 @@
+        });
+      }
+    } catch (error) {
+      toastRef?.current?.showToast({
```
I feel a utility for this would be excessive, this is single line access to the toast.

abretonc7s commented on app/components/UI/Perps/Views/PerpsOrderView/PerpsOrderView.tsx:
```diff
@@ -0,0 +1,1037 @@
+                    <Text variant={TextVariant.BodyLGMedium}>
+                      {orderForm.limitPrice
+                        ? formatPrice(orderForm.limitPrice)
+                        : 'Set price'}
```
I keep going back and forth on internationalizing or waiting for the screens to be finalized and fix everything at once...

abretonc7s committed:
feat: pr comments

abretonc7s committed:
formatting

abretonc7s commented on app/components/UI/Perps/Views/PerpsOrderView/PerpsOrderView.tsx:
```diff
@@ -0,0 +1,1037 @@
+  const cachedAccountState = usePerpsAccount();
+
+  // Get real HyperLiquid USDC balance
+  const availableBalance = parseFloat(
+    cachedAccountState?.availableBalance?.toString() || '0',
+  );
```
I think we can have a separate skeleton, having 0 at least prevent parsing errors

abretonc7s committed:
improved margin computation as per hl protocol

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
fix: unit tests

abretonc7s committed:
merge latest main

abretonc7s committed:
fix: infinite rendering in liquidatino price

abretonc7s committed:
formatting

abretonc7s committed:
pr comments

abretonc7s committed:
cleanup

gambinish committed:
Merge branch 'main' into metamask-mobile-perps-order

gambinish committed:
chore: Init PerpsTPSLBottomSheet.test.tsx

gambinish committed:
chore: PerpsTPSLBottomSheet.test.tsx

gambinish committed:
chore: Init PerpsLeverageBottomSheet

gambinish committed:
chore: PerpsLeverageBottomSheet.test.tsx

gambinish committed:
chore: init PerpsLimitPriceBottomSheet

gambinish committed:
chore: PerpsLimitPriceBottomSheet.test.tsx

gambinish committed:
chore: PerpsSlider.test.tsx

nickewansmith reviewed (changes requested):
This is looking good generally. I found a few things during e2e testing that we might want to look at before merging.

I am unable to edit TP/SL from the positions listing view, although I am able to set it in the actual order that is placed and see it in the success screen. It also does not show in the positions screen though TP/SL was set.

<img width="364" height="223" alt="Screenshot 2025-07-22 at 1 46 46 PM" src="https://github.com/user-attachments/assets/2641869e-9d92-4a37-937a-42b7335c8944" />

From this positions list view, if I click on a position and go to the position details screen, I am not able to edit TP/SL.

From the positions list, if I click the edit TP/SL button, the popup does show but upon setting values, the values do not persist. Also, the popup screen for the editing TP/SL shows underneath existing positions unless clicking on the edit TP/SL button for the last position listed

https://github.com/user-attachments/assets/921cfbff-5b21-4ec6-9372-0f286548e897

gambinish committed:
chore: Init usePerpsPaymentTokens.test.ts

gambinish committed:
chore: usePerpsPaymentTokens.test.ts

gambinish committed:
chore: Remove test

gambinish committed:
chore: PerpsOrderTypeBottomSheet.test.tsx

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
tpsl

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
full tpsl flow

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
pr comments

nickewansmith reviewed (dismissed):
Tested the new changes and It looks good to me! We may need a few more tests to pass coverage. 🤞🏽

abretonc7s committed:
manual input and order flow success

abretonc7s committed:
improved keypad order value

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
cleanup

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
coverage

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into metamask-mobile-perps-order

abretonc7s committed:
coverage

gambinish committed:
chore: Update unit tests, lint

gambinish committed:
chore: Update PerpsOrderView lint

gambinish committed:
chore: Cleanup unit tests

gambinish committed:
chore: More cleanup to PerpsOrderView test

gambinish committed:
chore: Bump coverage on error handling in HyperLiquidProvider

gambinish committed:
Merge branch 'main' into metamask-mobile-perps-order

gambinish committed:
Merge branch 'main' into metamask-mobile-perps-order

gambinish committed:
Merge branch 'main' into metamask-mobile-perps-order
